File: portal/portal/App_Code/DAO/ConteudoDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data;
using MySql.Data.MySqlClient;
using System.IO;
using MySql.Data.MySqlClient;
using System.IO;


namespace portal.App_Code.DAO
{
    public class ConteudoDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public Conteudo carregar(long pCodigo)
        {
            Conteudo obj = new Conteudo();
            MenuDAO menuDAO = new MenuDAO();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Conteudo where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);              
                    obj.DataPublicado = registros.GetDateTime("data_publicado").ToString("dd/MM/yyyy");
                    obj.Descritivo = registros["conteudo"].ToString();
                    obj.Keywords = registros["keywords"].ToString();
                    obj.Nome = registros["nome"].ToString();
                    obj.Status = (Conteudo.TipoStatus)Convert.ToUInt64(registros["status"]);
                    obj.Tipo = (Conteudo.Tipos)Convert.ToUInt64(registros["tipo"]);
                    obj.Titulo = registros["titulo"].ToString();
                    obj.Categorias = CategoriaDAO.carregarLista(obj.Codigo);

                    obj.MenuRelacionado = menuDAO.carregar(Convert.ToDouble(registros["codigo_menu"]));                            
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<Conteudo> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<Conteudo> lista = new List<Conteudo>();
            try
            {
                MenuDAO menuDAO = new MenuDAO();
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Conteudo where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                    if (i != pFiltros.Length - 1)
                    {
                        sql = sql + " and ";
                    }
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    Conteudo obj = new Conteudo();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.DataPublicado = registros.GetDateTime("data_publicado").ToString("dd/MM/yyyy");
                    obj.Descritivo = registros["conteudo"].ToString();
                    obj.Keywords = registros["keywords"].ToString();
                    obj.Nome = registros["nome"].ToString();
                    obj.Status = (Conteudo.TipoStatus)Convert.ToUInt64(registros["status"]);
                    obj.Tipo = (Conteudo.Tipos)Convert.ToInt64(registros["tipo"]);
                    obj.Titulo = registros["titulo"].ToString();
                    obj.Categorias = CategoriaDAO.carregarLista(obj.Codigo);
                    obj.MenuRelacionado = menuDAO.carregar(Convert.ToDouble(registros["codigo_menu"]));
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(Conteudo obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into Conteudo(nome, titulo, conteudo, status, keywords, tipo, data_publicado, codigo_menu) " +
                         " value('{0}','{1}','{2}','{3}','{4}','{5}','{6}',{7})";
                    sql = String.Format(sql, obj.Nome, obj.Titulo, obj.Descritivo, obj.Status, obj.Keywords, obj.Tipo,"sysdate", obj.MenuRelacionado.Codigo);
                    //grava cateforia    
                }
                else
                {
                    sql = "update Conteudo set nome='{0}', titulo='{1}', conteudo='{2}', keywords='{3}', codigo_menu={4} where codigo={5}";
                    sql = String.Format(sql, obj.Nome, obj.Titulo, obj.Descritivo, obj.Status, obj.Keywords, obj.Tipo, obj.MenuRelacionado.Codigo);
                }

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(Conteudo obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from Conteudo where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }

}

File: portal/portal/App_Code/DAO/ConteudoDAO.php
<?php
namespace portal\App_Code\DAO {
    
//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
use portal\App_Code\DAO\MenuDAO;
use portal\App_Code\DAO\CategoriaDAO;
use portal\App_Code\Conteudo;
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);

    try{
		require_once("MenuDAO.php");    
		require_once("CategoriaDAO.php");    
        require_once("/../Conteudo.php");        
        require_once("/../../connection.php");        
    }catch(Exception $e){
		require_once (dirname(__file__, 1) . "\MenuDAO.php");
		require_once (dirname(__file__, 1) . "\CategoriaDAO.php");
		require_once (dirname(__file__, 2) . "\Conteudo.php");
		require_once (dirname(__file__, 3) . "\connection.php");
    }
    
 class ConteudoDAO{
//INSERIR----------------------------------------------------------------------------------------------------------------------------------------
    function inserirConteudo($Conteudo){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into conteudo (nome, titulo,conteudo,status,keywords,tipo,data_publicado,codigo_menu) values (' " .
                 $Conteudo->getNome() . " ' , ' " .
                 $Conteudo->getTitulo() . " ' , ' " .
                 $Conteudo->getDescritivo() . " ' ,  " .
                 $Conteudo->getTipoStatus() . "  , ' " .
                 $Conteudo->getKeywords() . " ' ,  " .
                 $Conteudo->getTipo() . " , '" .
                 $Conteudo->getDataPublicado() . " ' , " . 
				 $Conteudo->getMenuRelacionado() . " ) ";
                 #$Conteudo->getMenuRelacionado()->getCodigo() . " ) ";
                 mysqli_query($con,$sql);
				 echo "Inserção com sucesso!";
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);	
	}

//ALTERAR---------------------------------------------------------------------------------------------------------------------------------------
    function alterarConteudo($Conteudo){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update conteudo set " . 
                "nome = ' " . $Conteudo->getNome() . " ' , " .
                "titulo = '" . $Conteudo->getTitulo() . " ' , " .
                "conteudo = '" . $Conteudo->getDescritivo() . " ' , " .
                "status = " . $Conteudo->getTipoStatus() . " , " .
                "keywords = '" . $Conteudo->getKeywords() . " ' , " . 
                "tipo = " . $Conteudo->getTipo() . " , " .
                "data_publicado = '" . $Conteudo->getDataPublicado() . " ' , " .
                "codigo_menu = " . $Conteudo->getMenuRelacionado() . #->getCodigo() .
                " where codigo = " . $Conteudo->getCodigo();
                mysqli_query($con,$sql);
				echo "Dados alterados com sucesso";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }
    mysqli_close($con);	
    }

//CARREGAR OBJETO CONTEUDO-----------------------------------------------------------------------------------------------------------------------
    function carregarConteudo ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from conteudo where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
						$contents = new Conteudo();
						$contents->setConteudo(($carga["codigo"]),
											   ($carga["nome"]),
											   ($carga["titulo"]),
											   ($carga["conteudo"]),
											   ($carga["status"]),
											   ($carga["keywords"]),
											   ($carga["tipo"]),
											   ($carga["data_publicado"]),
											   ($carga["codigo_menu"]));
						mysqli_close($con);
						return $contents;
					}else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;
            }
        mysqli_close($con);
    }
//------------------------------------------------------------------------------------------------------------------	
    function carregarConteudoComArray ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from conteudo where codigo = " . $codigo;
                
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
						$contents = new Conteudo();
						$contents->setConteudo(($$carga["codigo"]),
											   ($carga["nome"]),
											   ($carga["titulo"]),
											   ($carga["conteudo"]),
											   ($carga["status"]),
											   ($carga["keywords"]),
											   ($carga["tipo"]),
											   ($carga["data_publicado"]),
											   (carregarMenu($carga["codigo_menu"])),//Trazer retorno do objeto Menu.
                                               (carregarListaCategoriaCod($codigo))); //Retorna lista de objetos Categoria
						mysqli_close($con);
						return $contents;
					}else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;
            }
        mysqli_close($con);
    }

//CARREGAR LISTA DE OBJETOS CONTEÚDO----------------------------------------------------------------------------------------------------------
     function carregarListaConteudo($qtdOpcoes, $ordem){
            $sql = "select * from conteudo where ";   
            
            for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++){
                $sql = $sql . $qtdOpcoes[$i];
                    if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                    } 
            }
         
            $sql = $sql . " order by " . $ordem;
         
            try{
                $con = BD_AbrirConexao();
                $gargantula = mysqli_query($con,$sql);                
                    if($carga = mysqli_fetch_array($gargantula)){
                        $listaConteudo = [];
                        while($carga = mysqli_fetch_array($gargantula)){
						$contents = new Conteudo();
						$contents->setConteudo(($carga["codigo"]),
											   ($carga["nome"]),
											   ($carga["titulo"]),
											   ($carga["conteudo"]),
											   ($carga["keywords"]),
											   ($carga["tipo"]),
											   ($carga["data_publicado"]),
											   ($carga["status"]),
											   (carregarMenu($carga["codigo_menu"])),//Trazer retorno do objeto Menu.
                                               (carregarListaCategoriaCod($codigo))); //Retorna lista de objetos Categoria
                              $listaConteudo[] = $contents;
                         }
                        mysqli_close($con);
                        return $listaConteudo;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;           
            }
mysqli_close($con);
     }
                                        
//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarConteudo($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from conteudo where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
				echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;
            }
        mysqli_close($con);
    }
  }
}
?>